js/app/historyStore.js

```diff
@@ -91,9 +92,11 @@ const HistoryStore = new Lang.Class({
         Dispatcher.get_default().register((payload) => {
             switch(payload.action_type) {
                 case Actions.HISTORY_BACK_CLICKED:
+                    this.get_previous_item().entry_point = EntryPoints.NAV_BUTTON_CLICKED;
```

Comment:
The same item can cause more than one set of metrics if you go back to it, and the second set of metrics will have a different entry point. So to preserve the immutability of history items I think we'd have to decouple the entry point data from the history item, which would get messy.